File: backend/app/schemas/mrp.py
from datetime import datetime

from pydantic import BaseModel, Field


class MrpRunIn(BaseModel):
    scope: str = Field(default="all", max_length=255, description="运算范围：all 或逗号分隔的订单ID")
    order_ids: str | None = Field(default=None, description="逗号分隔的订单ID列表")


class MrpConvertIn(BaseModel):
    supplier_id: int = Field(ge=1)


class MrpDemandOut(BaseModel):
    id: int
    run_id: int
    sku_id: int
    sku_code: str | None = None
    sku_name: str | None = None
    required_qty: int
    in_stock_qty: int
    on_order_qty: int
    shortage_qty: int
    suggestion: str | None = None
    remark: str | None = None


class MrpRunOut(BaseModel):
    id: int
    code: str
    status: str
    scope: str
    run_at: datetime
    result_summary: str | None = None
    error_message: str | None = None
    created_by: int | None = None
    created_at: datetime


class MrpRunDetailOut(BaseModel):
    run: MrpRunOut
    demands: list[MrpDemandOut]

File: frontend-admin-pro/src/api/mrp.ts
import { http } from '@/utils/http'

export type MrpRunOut = {
  id: number
  code: string
  status: string
  scope: string
  run_at: string
  result_summary: string | null
  error_message: string | null
  created_by: number | null
  created_at: string
}

export type MrpDemandOut = {
  id: number
  run_id: number
  sku_id: number
  sku_code: string | null
  sku_name: string | null
  required_qty: number
  in_stock_qty: number
  on_order_qty: number
  shortage_qty: number
  suggestion: string | null
  remark: string | null
}

export function runMrp(scope = 'all', orderIds?: string) {
  return http.post<MrpRunOut>('/admin/mrp/run', null, { params: { scope, order_ids: orderIds } })
}

export function listMrpRuns(offset = 0, limit = 50) {
  return http.get<MrpRunOut[]>('/admin/mrp/runs', { params: { offset, limit } })
}

export function getMrpRunDetail(runId: number) {
  return http.get<{ run: MrpRunOut; demands: MrpDemandOut[] }>(`/admin/mrp/runs/${runId}`)
}

export function convertMrpToPurchaseOrder(runId: number, supplierId: number) {
  return http.post<{ purchase_order_id: number; code: string }>(`/admin/mrp/runs/${runId}/convert`, null, {
    params: { supplier_id: supplierId },
  })
}
